Fix html dependency parser to handle automatically frontend://

caalador opened the pull request:
Parser now also handles any dependency filters that there
may be for instance productionMode filter.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/flow/3646)
<!-- Reviewable:end -->

denis-anisimov commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73g0044ac04peiCysr:-L73g0044ac04peiCyss:b-zft8ym) (no related file):*
Is there an issue for that ?
I mean : is it important to exclude the imports that should be filtered out?
I've been thinking to add these filtering functionality from the very beginning but then decided do not do that.
The reason is: they will be indirectly imported anyway implicitly from the `HtmlImport` regardless of the added URLs via this dependency parser.
So I'm not sure it's worth to add this filtering logic....

---

*[flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java, line 111 at r1](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73fZv2PFPQLl1os33L:-L73fZv2PFPQLl1os33M:bhtevp0) ([raw file](https://github.com/vaadin/flow/blob/1b0014d62765dffe209e1288f8c39cfc15aab53b/flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java#L111)):*
> VaadinSession.getCurrent()

`session`

---

*[flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java, line 116 at r1](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73h4VSstWQMAkGSKqj:-L73h4VSstWQMAkGSKqk:b17ipa7) ([raw file](https://github.com/vaadin/flow/blob/1b0014d62765dffe209e1288f8c39cfc15aab53b/flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java#L116)):*
> new ArrayList<>(dependencyFilters)

is it required to make a copy each time ?

---

*[flow-server/src/test/java/com/vaadin/flow/component/internal/HtmlDependencyParserTest.java, line 51 at r1](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73j4sqhsAgNk-7J-Zj:-L73j4sqhsAgNk-7J-Zk:b9lm7x4) ([raw file](https://github.com/vaadin/flow/blob/1b0014d62765dffe209e1288f8c39cfc15aab53b/flow-server/src/test/java/com/vaadin/flow/component/internal/HtmlDependencyParserTest.java#L51)):*
> Mockito.when(service.getDependencyFilters())
>                 .thenReturn(Collections.emptyList());

Don't see tests for non-empty dependency filters.
Do you have them and I'm just blind?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3646)*
<!-- Sent from Reviewable.io -->

caalador commented:
Review status: 3 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73g0044ac04peiCysr:-L73xkksgPvedDD6OrzK:b-ar23z5) (no related file):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Is there an issue for that ?
I mean : is it important to exclude the imports that should be filtered out?
I've been thinking to add these filtering functionality from the very beginning but then decided do not do that.
The reason is: they will be indirectly imported anyway implicitly from the `HtmlImport` regardless of the added URLs via this dependency parser.
So I'm not sure it's worth to add this filtering logic....
</blockquote></details>

Removed dependency filtering.
dropped log level so production mode doesn't log as it will never find the resource.

---

*[flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java, line 111 at r1](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73fZv2PFPQLl1os33L:-L73xtRvTbk20MAdtRh4:b-hfpv0l) ([raw file](https://github.com/vaadin/flow/blob/1b0014d62765dffe209e1288f8c39cfc15aab53b/flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java#L111)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

> VaadinSession.getCurrent()

`session`
</blockquote></details>

removed

---

*[flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java, line 116 at r1](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73h4VSstWQMAkGSKqj:-L73mmF_IvG_H-xSaayO:b-xkfne7) ([raw file](https://github.com/vaadin/flow/blob/1b0014d62765dffe209e1288f8c39cfc15aab53b/flow-server/src/main/java/com/vaadin/flow/component/internal/HtmlDependencyParser.java#L116)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

> new ArrayList<>(dependencyFilters)

is it required to make a copy each time ?
</blockquote></details>

removed.

It's what the defaultTemplateParser does. 
I guess it's not needed as the filter always returns something that may be the list given in or a new list from the inside.

---

*[flow-server/src/test/java/com/vaadin/flow/component/internal/HtmlDependencyParserTest.java, line 51 at r1](https://reviewable.io:443/reviews/vaadin/flow/3646#-L73j4sqhsAgNk-7J-Zj:-L73xxOKQlXN4amuL6c-:b-hfpv0l) ([raw file](https://github.com/vaadin/flow/blob/1b0014d62765dffe209e1288f8c39cfc15aab53b/flow-server/src/test/java/com/vaadin/flow/component/internal/HtmlDependencyParserTest.java#L51)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

> Mockito.when(service.getDependencyFilters())
>                 .thenReturn(Collections.emptyList());

Don't see tests for non-empty dependency filters.
Do you have them and I'm just blind?
</blockquote></details>

removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3646)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3646)*
<!-- Sent from Reviewable.io -->
